experimental/devices/ina219/ina219.go

```diff
@@ -0,0 +1,192 @@
+// Copyright 2018 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+package ina219
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"sync"
+
+	"periph.io/x/periph/conn/i2c"
+	"periph.io/x/periph/conn/mmr"
+	"periph.io/x/periph/conn/physic"
+)
+
+// Opts holds the configuration options.
+//
+// Slave Address
+//
+// Depending which pins the A1, A0 pins are connected to will change the slave
+// address. Default configuration is address 0x40 (both pins to GND). For a full
+// address table see datasheet.
+type Opts struct {
+	Address       int
+	SenseResistor physic.ElectricResistance
+	MaxCurrent    physic.ElectricCurrent
+}
+
+// DefaultOpts is the recommended default options.
+var DefaultOpts = Opts{
+	Address:       0x40,
+	SenseResistor: 100 * physic.MilliOhm,
+	MaxCurrent:    3200 * physic.MilliAmpere,
+}
+
+// New opens a handle to an ina219 sensor.
+func New(bus i2c.Bus, opts *Opts) (*Dev, error) {
+
+	i2cAddress := DefaultOpts.Address
+	if opts.Address != 0 {
```

Comment:
It's fine to force users to pass valid value and not 0, as there's DefaultOpts. Same for line 50, 58, which would simplify this code a bit.